analyzers/src/SonarAnalyzer.Common/Rules/DebuggerDisplayUsesExistingMembersBase.cs

```diff
@@ -0,0 +1,89 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2023 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Text.RegularExpressions;
+
+namespace SonarAnalyzer.Rules;
+
+public abstract class DebuggerDisplayUsesExistingMembersBase<TAttributeSyntax, TSyntaxKind> : SonarDiagnosticAnalyzer<TSyntaxKind>
+    where TAttributeSyntax : SyntaxNode
+    where TSyntaxKind : struct
+{
+    private const string DiagnosticId = "S4545";
+
+    private readonly Regex noQuotesModifierExpressionRegex = new(@",\s*nq\s*$", RegexOptions.None, RegexConstants.DefaultTimeout);
```

Comment:
Nitpick: $ symbolizes the end of the string or the end of the line in a regular expression. Why is it needed here?

Reply:
This regex is targeting a single evaluate expression, surrounded by braces: `blah {expr1} blah {expr2} blah`
Each of these expressions can be arbitrarily complex, and potentially contain the string `,nq`.
For example `{",nq",nq}`, which should make the debugger print `",nq"` without quote symbols around the string.
By using `$` I want to make sure that I am selecting the nq immediately preceding the closing brace symbol, and nothing else.